src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -198,23 +201,76 @@ def load_arguments(self, _):
+        c.argument('template_uri', arg_type=deployment_template_uri_type)
+        c.argument('parameters', arg_type=deployment_parameters_type)
 
     with self.argument_context('deployment create') as c:
-        c.argument('deployment_name', options_list=['--name', '-n'], required=False,
-                   help='The deployment name. Default to template file base name')
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
 
     with self.argument_context('deployment validate') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
 
-    with self.argument_context('deployment operation show') as c:
+    with self.argument_context('deployment operation') as c:
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
 
+    with self.argument_context('deployment sub') as c:
+        c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
+
+    with self.argument_context('deployment sub create') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment sub validate') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment group') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type, completer=get_resource_group_completion_list, required=True)
+        c.argument('mode', arg_type=get_enum_type(DeploymentMode, default='incremental'), help='Incremental (only add resources to resource group) or Complete (remove extra resources from resource group)')
+        c.argument('rollback_on_error', nargs='?', action=RollbackAction,
+                   help='The name of a deployment to roll back to on error, or use as a flag to roll back to the last successful deployment.')
+
+    with self.argument_context('deployment group create') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+        c.argument('aux_subscriptions', nargs='*', options_list=['--aux-subs'],
+                   help='Auxiliary subscriptions which will be used during deployment across tenants.')
+
+    with self.argument_context('deployment group validate') as c:
+        c.argument('deployment_name', arg_type=deployment_create_name_type)
+        c.argument('handle_extended_json_format', arg_type=extended_json_format_type)
+
+    with self.argument_context('deployment mg') as c:
```

Comment:
`deployment mg` -> `deployment management-group`  Maybe the naming looks clearer and more elegant?

Reply:
our only concern is the number of characters to type out `management-group` especially when there is no autocomplete for cli (at least on windows from what I can find). MG is a fairly well-known acronym. Thoughts?

Reply:
@alex-frankel Personally, I mainly think `mg` doesn't look intuitive enough. However if `mg` is used in other commands, and the user already knows this, I think `mg` is fine.

Reply:
"mg" is actually something new.. I don't think other commands use it today. @alex-frankel what do you think?

Reply:
I didn't mean to suggest it is used in other cli commands, but it is a commonly used acronym in customer conversations, so I'm not worried about it

Reply:
Ok, I think `mg` is fine.